service/resource/configmap/resource.go

```diff
@@ -193,39 +223,18 @@ func (r *Resource) GetDeleteState(ctx context.Context, obj, currentState, desire
 	return configMapsToDelete, nil
 }
 
-func (r *Resource) GetUpdateState(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, interface{}, interface{}, error) {
+func (r *Resource) newUpdateChange(ctx context.Context, obj, currentState, desiredState interface{}) (interface{}, error) {
 	customObject, err := key.ToCustomObject(obj)
 	if err != nil {
-		return nil, nil, nil, microerror.Mask(err)
+		return nil, microerror.Mask(err)
 	}
 	currentConfigMaps, err := toConfigMaps(currentState)
 	if err != nil {
-		return nil, nil, nil, microerror.Mask(err)
+		return nil, microerror.Mask(err)
 	}
 	desiredConfigMaps, err := toConfigMaps(desiredState)
 	if err != nil {
-		return nil, nil, nil, microerror.Mask(err)
-	}
-
-	var configMapsToCreate interface{}
-	{
-		configMapsToCreate, err = r.GetCreateState(ctx, obj, currentState, desiredState)
-		if err != nil {
-			return nil, nil, nil, microerror.Mask(err)
-		}
-	}
-
-	var configMapsToDelete []*apiv1.ConfigMap
-	{
-		r.logger.Log("cluster", key.ClusterID(customObject), "debug", "finding out which config maps have to be deleted")
-
-		for _, currentConfigMap := range currentConfigMaps {
-			if !containsConfigMap(desiredConfigMaps, currentConfigMap) {
-				configMapsToDelete = append(configMapsToDelete, currentConfigMap)
-			}
-		}
-
-		r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("found %d config maps that have to be deleted", len(configMapsToDelete)))
```

Comment:
When I implemented this I considered deletion of resources during updates. I guess this was never really useful. So it should be ok to get rid of it and implement it in case we need it. This was part of the learning curve on my side.